app/js/enhancedDate.js

```diff
@@ -1,5 +1,71 @@
+var enhancedDate = (function() {
+	var dateIsSet;
+	var enhancedDateObject = {
+		setDate: setDate,
+		getDate: getDate,
+		getDayName: getDayName,
+		getMonthName: getMonthName,
+		isFuture: isFuture,
+		isToday: isToday
+	};
+	return enhancedDateObject;
+
+
+	function setDate(date) {
+		if (!date) {dateIsSet = new Date(); return;}
+		if (!(date instanceof Date)) {console.warn('sorry, not a date'); return;}
```

Comment:
setDate should be able to take either a Date object or a millisecond value - if nothing is passed, it should default to creating a new Date based on "now"